std/string.d

```diff
@@ -1815,6 +1815,268 @@ ptrdiff_t lastIndexOfNeither(Char,Char2)(const(Char)[] haystack,
     );
 }
 
+version(randomized_unittest_benchmark) unittest
+{
+    import std.internal.test.randomized_unittest_benchmark;
+    import std.range : chain, iota;
+    import std.conv : to;
+    import std.stdio : writefln;
+    import std.algorithm : equal;
+
+    enum charsToSearch = chain(iota(0x21, 0x7E), iota(0xA1, 0x1FF))
+        .to!dstring();
+    enum size_t len = charsToSearch.length;
+
+    immutable rounds = 20000;
+    auto rnd = Random(1);
+
+    foreach (S; TypeTuple!(string, wstring, dstring))
```

Comment:
You know about the backend optimizer issue with long function bodies?
[Issue 7157 – Optimiser is O(n^2) w.r.t. function length](https://issues.dlang.org/show_bug.cgi?id=7157)
You better duplicate the foreach for each nested scope.